model/adjuster/otel_tag_test.go

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) 2023 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package adjuster
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	semconv "go.opentelemetry.io/otel/semconv/v1.21.0"
+
+	"github.com/jaegertracing/jaeger/model"
+)
+
+func TestOTelTagAdjuster(t *testing.T) {
+	testCases := []struct {
+		description string
+		span        *model.Span
+		expected    *model.Span
+	}{
+		{
+			description: "span with otel library tags",
+			span: &model.Span{
+				Tags: model.KeyValues{
+					model.String("random_key", "random_value"),
+					model.String(string(semconv.OTelLibraryNameKey), "go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp"),
+					model.String(string(semconv.OTelLibraryVersionKey), "0.45.0"),
+				},
+				Process: &model.Process{
+					Tags: model.KeyValues{},
+				},
+			},
+			expected: &model.Span{
+				Tags: model.KeyValues{
+					model.String("random_key", "random_value"),
+				},
+				Process: &model.Process{
+					Tags: model.KeyValues{
+						model.String(string(semconv.OTelLibraryNameKey), "go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp"),
+						model.String(string(semconv.OTelLibraryVersionKey), "0.45.0"),
+					},
+				},
+			},
+		},
+		{
+			description: "span without otel library tags",
+			span: &model.Span{
+				Tags: model.KeyValues{
+					model.String("random_key", "random_value"),
+				},
+				Process: &model.Process{
+					Tags: model.KeyValues{},
+				},
+			},
+			expected: &model.Span{
+				Tags: model.KeyValues{
+					model.String("random_key", "random_value"),
+				},
+				Process: &model.Process{
+					Tags: model.KeyValues{},
+				},
+			},
+		},
+	}
+	for _, testCase := range testCases {
+		t.Run(testCase.description, func(t *testing.T) {
+			trace := &model.Trace{
+				Spans: []*model.Span{testCase.span},
+			}
+			trace, err := OTelTagAdjuster().Adjust(trace)
+			assert.NoError(t, err)
+			assert.Equal(t, testCase.expected.Tags, trace.Spans[0].Tags)
+			assert.Equal(t, testCase.expected.Process.Tags, trace.Spans[0].Process.Tags)
```

Comment:
Hmm.. do you have in mind on how to test this? 🤔 I actually can't find out how to ensure this, or can you give me an actual test case?

Reply:
Pretty much like in the example I pointed to. Before L83 store a reference to `beforeTags := testCase.span.Tags` slice. After Adjust is called:
```
newTag := model.String("new tag", "new value")
beforeTags[0] = newTag
assert.Equal(t, newTag, testCase.span.Tags[0])
```
The assert confirms that `beforeTags` and `testCase.span.Tags` still point to the same underlying slice.
You can even add
```
assert.NotEqual(t, beforeTags, testCase.span.Tags)
```

Reply:
Ahh, now I get what you meant. I've added the test, but I don't think `assert.NotEqual(t, beforeTags, testCase.span.Tags)` works because `beforeTags` and `testCase.span.Tags` point to the same underlying slice which resulting they always equal.

Reply:
they point to the same underlying _array_, but the slices are different - the one after the adjustment is shorter

Reply:
Ohh right..